### PR TITLE
Fixes various bow issues and includes new sounds for the bow

### LINKED ISSUE #75803: Bows have the low ammo clicking sound when fired
## Reproduction:

Bows play the 'low ammo' clicking sound every time they fire, which isn't really realistic for simple ranged weapon.

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

fixes https://github.com/tgstation/tgstation/issues/75803

Fixes various bow issues;
Bows no longer can be loaded while in a slot other than your inactive hand. 

Dropping bows or putting them into a slot OTHER than your hand now drops the arrow, as intended. No longer hardcoded to do this only if it goes on your back.

Divine bows can fit in the chaplain suit storage, like most other bulky null rods. As a result, has a suit storage sprite.

Drawing, loading and firing the bow now plays sounds instead of being entirely silent. New sounds adapted from here:
https://freesound.org/people/SonoFxAudio/sounds/649337/ <- bow_draw.ogg
https://freesound.org/people/SonoFxAudio/sounds/649334/ <- bow_fire.ogg

cleans up the code a bit as well.

## Why It's Good For The Game

*thunk*

Also because I think the bow being completely silent is utterly jarring and it had some really weird hardcoded issues.

## Changelog
:cl:
fix: Bows properly update their ammo when stored, dropped and stowed. You cannot load a bow without holding it in your inactive hand. You have to hold it to keep the bow loaded.
qol: You can remove an arrow from the bow with alt+click, much like other ballistic weapons.
fix: The divine bow can be properly stored in the chaplain armor's suit storage, much like all other bulky null rods.
sound: Bows have sounds for drawing, firing and loading.
/:cl:

### CODE REVIEW

**Comment by Sealed101** [comment]
there's also #75448 that's bow related :^)

**Comment by necromanceranne** [comment]
bow related but it looks more like an issue with the component.

File: code/modules/jobs/job_types/chaplain/chaplain_costumes.dm [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
+	allowed = list(
+		/obj/item/book/bible,
+		/obj/item/nullrod,
+		/obj/item/reagent_containers/cup/glass/bottle/holywater,
+		/obj/item/storage/fancy/candle_box,
+		/obj/item/flashlight/flare/candle,
+		/obj/item/tank/internals/emergency_oxygen,
+		/obj/item/tank/internals/plasmaman,
+		/obj/item/gun/ballistic/bow/divine
+	)
```
Comment (MrMelbert) [inline code/modules/jobs/job_types/chaplain/chaplain_costumes.dm:13]: Make this a global list both suits can use (`Initialize() allowed = GLOB.chaplain_suit_items`)

**Review by Wallemations — APPROVED** [review]
bow suit sprite looks good

File: code/modules/jobs/job_types/chaplain/chaplain_costumes.dm [hunk 2]
```diff
@@ -24,7 +28,16 @@
...
```
Comment (san7890) [inline code/modules/jobs/job_types/chaplain/chaplain_costumes.dm:40]: you just made that fancy global list why are you not using that again here

**Review by san7890 — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/ballistic/bows/_bow.dm [hunk 3]
```diff
@@ -20,13 +20,29 @@
...
+		magazine.get_round(keep = FALSE)
+
+		var/obj/item/ammo_casing/caseless/arrow/our_arrow = chambered
+		user.put_in_hands(our_arrow)
+
+		drawn = FALSE
+		chambered = null
+		update_appearance()
+		return
+	return ..()
```
Comment (san7890) [inline code/modules/projectiles/guns/ballistic/bows/_bow.dm:44]: early return here instead, also isnull is faster *mumble mumble*
```suggestion
	if(isnull(chambered))
		return ..()

	chambered.forceMove(drop_location())
	magazine.get_round(keep = FALSE)

	var/obj/item/ammo_casing/caseless/arrow/our_arrow = chambered
	user.put_in_hands(our_arrow)

	drawn = FALSE
	chambered = null
	update_appearance()
```

**Review by san7890 — CHANGES_REQUESTED** [review]
last thing

File: code/modules/projectiles/guns/ballistic/bows/_bow.dm (hunk 3 above)
Comment (necromanceranne) [reply code/modules/projectiles/guns/ballistic/bows/_bow.dm:44]: Sure thing boss

**Pull request merged**